apps/advisor/pkg/app/checkscheduler/checkscheduler.go

```diff
@@ -89,14 +90,58 @@ func (r *Runner) Run(ctx context.Context) error {
 	// We still need the context to eventually be cancelled to exit this function
 	// but we don't want the requests to fail because of it
 	ctxWithoutCancel := context.WithoutCancel(ctx)
-	lastCreated, err := r.checkLastCreated(ctxWithoutCancel, logger)
+
+	// Determine namespaces based on StackID or OrgID
+	namespaces, err := checks.GetNamespaces(ctxWithoutCancel, r.stackID, r.orgService)
+	if err != nil {
+		return fmt.Errorf("failed to get namespaces: %w", err)
+	}
+
+	logger.Debug("Scheduling checks", "namespaces", len(namespaces))
+
+	// Start a goroutine for each namespace
+	var wg sync.WaitGroup
+	errChan := make(chan error, len(namespaces))
+
+	for _, namespace := range namespaces {
+		wg.Add(1)
+		go func(ns string) {
```

Comment:
Any concern about customers with many orgs? I have no idea how common that is, but there are many folks operating Grafana as a service and using orgs per customer.

At the very least, are there any optimizations needed if we have > 20 schedulers running at once for example? You mention `we need to evaluate each org independently` but curious why the namespacing logic couldn't be done per goroutine (IE have 1 goroutine at most whether on-prem or cloud)

Reply:
Take a look to the alternative now, the changes are:

 - The next evaluation time is based in the oldest check among all orgs.
 - Only for the orgs that have their checks older than the evaluation interval we create the checks.
 
The caveat is that some orgs may see longer wait times (not exactly the evaluation interval) but it's not a big deal.

Not sure if I'm missing something.